community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java

```diff
@@ -615,14 +616,17 @@ public void updateDynamicSetting( String setting, String newValue ) throws Illeg
                 validator.validate( newEntry, ignore -> {} ); // Throws if invalid
             }
 
-            oldValue = getDefaultValueOf( setting );
+            oldValue = getConfiguredValueOf( setting );
 
-            params.put( setting, newValue );
+            if ( params.put( setting, newValue ) == null )
```

Comment:
this assumption is not really correct since it can be the case that some property was reset to be null and that is not a default value of right?

Reply:
Due to the way lookup works, `null` is treated as the default value. So technically you cannot set `null` as a value and have the setting return that unless the default value is `null`. See `org.neo4j.kernel.configuration.Settings.DefaultSetting#apply`.